scalameta/contrib/shared/src/main/scala/scala/meta/contrib/ScaladocParser.scala

```diff
@@ -43,6 +45,18 @@ object ScaladocParser {
     comment.content.map(parseRec)
   }
 
+  private[this] def generateHeadingParser(level: Int, f: (String) => DocToken): Parser[DocToken] = {
+    val headingSimbols = "=" * level
+    P(
+      // Code block start
+      headingSimbols
+      // Heading description
+        ~ ((AnyChar ~ !"=").rep ~ AnyChar).!.map(f(_))
```

Comment:
Did you try `flatMap`? http://www.lihaoyi.com/fastparse/#FlatMap We should avoid exceptions,  parsers can report errors at positions.